docs/docs/how-to/dbauth-passwordless.md

```diff
@@ -0,0 +1,644 @@
+  updateUser(id: Int!, input: UpdateUserInput!): User! @requireAuth
+  deleteUser(id: Int!): User! @requireAuth
+  generateToken(email: String!): userTokenResponse! @skipAuth
+}
+```
+
+### 4. Modify the auth function
+
+We need to consider how we want to limit the authentication.  I've added a expiration date to the token, so we'll need to check that.
+
+```js title="/api/src/functions/auth.js"
+// ... other functions
+const loginOptions = {
+  handler: async (user) =>{
+    let loginExpiresAt = new Date(user?.loginTokenExpiresAt)
+    let now = new Date()
+    let tokenExpired = loginExpiresAt < now
+    if(tokenExpired) throw 'Login token expired'
+    // if the user logged in with a token we need to break
+    // the token.  We'll do this by clearing the salt and
+    // expiration
+    // this will make the token a one-time use
+    let where = { id: user.id }
+    let data = {
+      loginTokenExpiresAt: null,
+      salt: null,
+    }
+    let result = db.user.update({ where, data })
+    return user
+  },
+  errors: {
+    // here I modified the following, feel free to modify the other messages
+    incorrectPassword: 'Incorrect token',
+  }
+}
+// we also need to update the signupOptions
+const signupOptions = {
+  handler: ({ username, hashedPassword, userAttributes }) => {
+    return db.user.create({
+      data: {
+        email: username,
+        loginToken: hashedPassword,
+        salt: null,
+        name: userAttributes.name,
+      }
+    })
+  }
+  // ... othter stuff
+}
+// and last we need to update the authFields
+const authHandler = new DbAuthHandler(event, context, {
+  db: db,
+  authModelAccessor: 'user',
+  authFields: {
+    id: 'id',
+    hashedPassword: 'loginToken',
+    salt: 'salt',
+    resetToken: 'resetToken',// we don't use this.
+    resetTokenExpiresAt: 'resetTokenExpiresAt',// we don't use this.
+  },
+  // ... other stuff
+})
+```
+
+As of right now, nothing works, lets fix that.
+
+### 5. Making the login form
+
+We need to make a form that will allow the user to enter their email address.
+
+Let's start with the generator.
+
+```bash
+yarn rw g component LoginPasswordlessForm
+```
+
+This created a component in `web/src/components/LoginPasswordlessForm/LoginPasswordlessForm.js`.  Let's update it.
+
+```jsx title="/web/src/components/LoginPasswordlessForm/LoginPasswordlessForm.js"
+import {
+  Form,
+  Label,
+  TextField,
+  PasswordField,
+  Submit,
+  FieldError,
+  useForm,
+} from '@redwoodjs/forms'
+import { navigate, routes, Link } from '@redwoodjs/router'
+import { MetaTags, useMutation } from '@redwoodjs/web'
+import { Toaster, toast } from '@redwoodjs/web/toast'
+const GENERATE_LOGIN_TOKEN = gql`
+  mutation generateLoginToken($email: String!) {
+    generateLoginToken(email: $email) {
+      message
+    }
+  }
+`
+
+const LoginPasswordlessForm = ({ setWaitingForCode, setEmail }) => {
+  const [generateLoginToken, { loading, error }] = useMutation(
```

Comment:
``{ loading, error }`` is not used. I don't think it's pertinent to assign unused variables.